### PR TITLE
(plugin) os::picos::snmp - Adding PICOS plugin 

### PR DESCRIPTION
**Pull request by lchrdn**
Adding PICOS from Pica8 plugin

### CODE REVIEW

File: os/picaos/snmp/mode/cpu.pm [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+sub manage_selection {
+    my ($self, %options) = @_;
+
+    my $oid_cpuUsage = '.1.3.6.1.4.1.35098.1.1.0';
+    my $oid_cpuTemperature = '.1.3.6.1.4.1.35098.1.6.0';
+
+    my $oids = [$oid_cpuUsage, $oid_cpuTemperature];
+    my $snmp_result = $options{snmp}->get_leef(oids => $oids, nothing_quit => 1);
+
+    $self->{global} = { prct_used => $snmp_result->{$oid_cpuUsage},
```
Comment (garnier-quentin) [inline os/picaos/snmp/mode/cpu.pm:76]: wrong indentation

File: os/picaos/snmp/mode/cpu.pm [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                perfdatas => [
+                    { label => 'cpu', template => '%.2f', unit => '%', min => 0, max => 100 },
+                ],
+            }
+        },
+        { label => 'temperature', nlabel => 'cpu.temperature.celsius', set => {
+                key_values => [ { name => 'temperature' } ],
+                output_template => 'CPU Temperature: %s C',
+                perfdatas => [
+                    { label => 'cpu', template => '%s', unit => 'C' },
```
Comment (garnier-quentin) [inline os/picaos/snmp/mode/cpu.pm:48]: label is useless with force perfdata

File: os/picaos/snmp/mode/cpu.pm [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    $self->{maps_counters_type} = [
+        { name => 'global', type => 0 },
+    ];
+    
+    $self->{maps_counters}->{global} = [
+        { label => 'cpu', nlabel => 'cpu.utilization.percentage', set => {
+                key_values => [ { name => 'prct_used' } ],
+                output_template => 'CPU Usage %.2f %%',
+                perfdatas => [
+                    { label => 'cpu', template => '%.2f', unit => '%', min => 0, max => 100 },
```
Comment (garnier-quentin) [inline os/picaos/snmp/mode/cpu.pm:40]: label is useless with force perfdata

File: os/picaos/snmp/mode/interfaces.pm [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+                    { template => '%.2f', unit => 'C', label_extra_instance => 1, instance_use => 'display' }
+                ]
+            }
+        }
+    ;
+}
+
+sub new {
+    my ($class, %options) = @_;
+    my $self = $class->SUPER::new(package => __PACKAGE__, %options);
```
Comment (garnier-quentin) [inline os/picaos/snmp/mode/interfaces.pm:79]: please set force perfdata

File: os/picaos/snmp/mode/temperature.pm [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+                perfdatas => [
+                    { label => 'chip', template => '%s', unit => 'C' },
+                ],
+            }
+        },
+        { label => 'cpu-temperature', nlabel => 'cpu.temperature.celsius', set => {
+                key_values => [ { name => 'cpu_temperature' } ],
+                output_template => 'CPU Temperature: %s C',
+                perfdatas => [
+                    { label => 'cpu', template => '%s', unit => 'C' },
```
Comment (garnier-quentin) [inline os/picaos/snmp/mode/temperature.pm:56]: label is useless

File: os/picaos/snmp/mode/temperature.pm [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+                ]
+            }
+        },
+        { label => 'chip-temperature', nlabel => 'chip.temperature.celsius', set => {
+                key_values => [ { name => 'chip_temperature' } ],
+                output_template => 'Chip Temperature: %s C',
+                perfdatas => [
+                    { label => 'chip', template => '%s', unit => 'C' },
```
Comment (garnier-quentin) [inline os/picaos/snmp/mode/temperature.pm:48]: label is useless

File: os/picaos/snmp/mode/interfaces.pm (hunk 4 above)
Comment (lchrdn) [reply os/picaos/snmp/mode/interfaces.pm:79]: oups

**Review by garnier-quentin — DISMISSED** [review]

**Review by garnier-quentin — DISMISSED** [review]

**Review by garnier-quentin — APPROVED** [review]

**Review by garnier-quentin — APPROVED** [review]

**Pull request merged**